### PR TITLE
fizzbuzz passing tests

### PR DESCRIPTION
**Pull request by lostybtw**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/fizz_buzz.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    while number <= iterations:
+        if number % fizz_number == 0:
+            out += "Fizz"
+        if number % buzz_number == 0:
+            out += "Buzz"
+        if out == "":
+            out = str(number)
+
+        print(out)
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:64]: Why are you printing `out`?

File: dynamic_programming/fizz_buzz.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      ...
+    ValueError: iterations must be done more than 0 times to play FizzBuzz
+    >>> fizz_buzz('a',"",5,3,5)
+    Traceback (most recent call last):
+      ...
+    ValueError: starting number must be and integer and be more than 0
+
+    """
+
+    if not type(iterations) == int:
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:45]:
```suggestion
    if not isisinstance(iterations, int):
```

File: dynamic_programming/fizz_buzz.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        raise ValueError("iterations must be defined as integers")
+    if not type(fizz_number) == int:
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:47]:
```suggestion
    if not isisinstance(fizz_number, int):
```

File: dynamic_programming/fizz_buzz.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        raise ValueError("fizz_number must be defined as an integer")
+    if not type(buzz_number) == int:
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:49]:
```suggestion
    if not isisinstance(buzz_number, int):
```

File: dynamic_programming/fizz_buzz.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        raise ValueError("buzz_number must be defined as an integer")
+    if not iterations >= 1:
+        raise ValueError("iterations must be done more than 0 times to play FizzBuzz")
+    if not type(number) == int or not number >= 1:
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:53]:
```suggestion
    if not isisinstance(number, int) or not number >= 1: 
```

File: dynamic_programming/fizz_buzz.py (hunk 1 above)
Comment (lostybtw) [reply dynamic_programming/fizz_buzz.py:64]: To Show FizzBuzz Output

Comment (lostybtw) [reply dynamic_programming/fizz_buzz.py:64]: Shall I make it return `out` ?

File: dynamic_programming/fizz_buzz.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    while number <= iterations:
+        if number % fizz_number == 0:
+            out += "Fizz"
+        if number % buzz_number == 0:
+            out += "Buzz"
+        if out == "":
+            out = str(number)
+
+        return out
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:64]: You're returning on the first iteration, you need to return at the end

File: dynamic_programming/fizz_buzz.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# https://en.wikipedia.org/wiki/Fizz_buzz#Programming
+
+
+def fizz_buzz(
+    number: int, out: str, iterations: int, fizz_number: int, buzz_number: int
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:5]: `out`, `fizz_numher` and `buzz_number` shouldn't be passed, as they are temp/const variables.

You could perhaps define ```py
FIZZ_NUMBER = 3
BUZZ_NUMBER = 5
```

File: dynamic_programming/fizz_buzz.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        if number % fizz_number == 0:
+            out += "Fizz"
+        if number % buzz_number == 0:
+            out += "Buzz"
+        if out == "":
+            out = str(number)
+
+        return out
+        number += 1
+        out = ""
```
Comment (CaedenPH) [inline dynamic_programming/fizz_buzz.py:67]:
```suggestion
        number += 1
        out = ""
return out
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Almost there

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**